src/Poseidon/EntitiesList.hs

```diff
@@ -39,63 +40,98 @@ instance Show SignedEntity where
 
 type SignedEntitiesList = [SignedEntity]
 
-isInclude :: SignedEntity -> Bool
-isInclude (Include _) = True
-isInclude _ = False
-
-isExclude :: SignedEntity -> Bool
-isExclude (Exclude _) = True
-isExclude _ = False
-
-fromSignedEntity :: SignedEntity -> PoseidonEntity
-fromSignedEntity (Include a) = a
-fromSignedEntity (Exclude a) = a
-
-entityIncludes :: SignedEntitiesList -> EntitiesList
-entityIncludes xs = map fromSignedEntity $ filter isInclude xs
-
-entityExcludes :: SignedEntitiesList -> EntitiesList
-entityExcludes xs = map fromSignedEntity $ filter isExclude xs
-
--- | A parser to parse entities
-parseSignedEntitiesList :: P.Parser SignedEntitiesList
-parseSignedEntitiesList = do 
-    eL <- P.sepBy parsePoseidonEntity (P.char ',' <* P.spaces)
-    P.optional (P.many (P.char ' ') >> comment)
-    return eL
-
-parseSignedEntitiesListFromFile :: P.Parser SignedEntitiesList
-parseSignedEntitiesListFromFile = do
-    concat <$> P.sepBy (P.try emptyLine <|> comment <|> parseSignedEntitiesList) P.newline
+-- A class to generalise signed and unsigned Entity Lists. Both have the feature that they can be used to filter individuals.
+class EntitySpec a where
+    indInfoConformsToEntitySpec :: [a] -> IndividualInfo -> Bool
```

Comment:
This is not specifically for this line, but more of a general observation: The definition of `IndividualInfo` seems to be less powerful than it could be. 
```haskell
data IndividualInfo = IndividualInfo { 
      indInfoName :: String
    , indInfoGroups :: [String]
    , indInfoPacName :: String
    }
```
I don't understand your logic deep down, but I feel this could be more precise. Maybe we should introduce dedicated `type`s for these categories which could then also be used in `PoseidonEntity` :thinking: 
```haskell
type PacName = String
type GroupName = String
type IndName = String

data IndividualInfo = IndividualInfo { 
      indInfoName :: PacName
    , indInfoGroups :: [GroupName]
    , indInfoPacName :: IndName
    }
```
Maybe it's too painful, but `indInfoGroups` could be a [`NonEmpty`](https://hackage.haskell.org/package/base-4.16.0.0/docs/Data-List-NonEmpty.html)?

Reply:
Hmm, but `type` is simply an Alias, not really a type. It doesn't provide any additional compiler-constraints. For example, even with these three type annotations, you can still do:
```
indName :: String
indName = "foo"
groupNames :: [String]
groupNames = ["group1", "group2"]
pacName :: String
pacName = "bar"
```

So, if you really want to make that more type-safe, you'd have to use a `newtype` or `data` declaration, which then would literally forbid supplying strings. I'm not sure whether that's the way to go. Semantically a group-name is literally a name, so a string, there is no additional semantic quality to it which would warrant a new-type, in my view.

But `type` only makes it look nicer, with no additional compiler guarantees whatsoever, I'm afraid, so I don't think that will help much.

Reply:
And regarding `NonEmpty`: Well, I'm curious how that would go. My understanding so far of `NonEmpty` is that while it's very nice in principle, it would have a pretty substantial cost downstream, because basically none of the typical list methods like concat and stuff would work. All these would have to replaced. Maybe that's not too hard to do, but remember that it would mean also changing this in the Janno definition, and therefore it would show up everywhere in the code-base. I'm not against it, just haven't done any research how this works.

Reply:
Hmm, very interesting, this `NonEmpty` suggestion. I just realised that that type is part of `base` and seems to support most important type classes (`Functor`, `Foldable`...), and all the necessary functions are there.

I think that's a very good suggestion. But I'm a bit anxious that it would require quite some code changes, since it would be already be part of Janno...

Reply:
Thanks for the explanation about `type`. That was a misconception of mine. Only really discovered `type` recently.

I'll open an issue to consider `NonEmpty` in future refactoring. Doesn't have to be part of this PR.